OpenDreamRuntime/DreamManager.Connections.cs

```diff
@@ -32,6 +50,119 @@ public sealed partial class DreamManager {
             _netManager.RegisterNetMessage<MsgLoadInterface>();
             _netManager.RegisterNetMessage<MsgAckLoadInterface>(RxAckLoadInterface);
             _netManager.RegisterNetMessage<MsgSound>();
+
+            var worldTopicAddress = new IPEndPoint(IPAddress.Loopback, _netManager.Port);
+            _sawmill.Debug($"Binding World Topic at {worldTopicAddress}");
+            _worldTopicSocket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp) {
+                ReceiveTimeout = 5000,
+                SendTimeout = 5000,
+                ExclusiveAddressUse = false,
+            };
+            _worldTopicSocket.Bind(worldTopicAddress);
+            _worldTopicSocket.Listen();
+            _worldTopicCancellationToken = new CancellationTokenSource();
+            _worldTopicListener = WorldTopicListener(_worldTopicCancellationToken.Token);
+        }
+
+        private void ShutdownConnectionManager() {
+            _worldTopicSocket!.Dispose();
+            _worldTopicCancellationToken!.Cancel();
+        }
+
+        private async Task WorldTopicListener(CancellationToken cancellationToken) {
+            async Task<string?> ParseByondTopic(Socket from) {
+                var buffer = new byte[2];
+                await from.ReceiveAsync(buffer, cancellationToken);
+                if (!buffer.SequenceEqual(ByondTopicHeaderRaw)) {
+                    if (buffer.SequenceEqual(ByondTopicHeaderEncrypted))
+                        _sawmill.Warning("Encrypted World Topic request is not implemented.");
+                    return null;
+                }
+
+                await from.ReceiveAsync(buffer, cancellationToken);
+                if (BitConverter.IsLittleEndian)
+                    buffer = buffer.Reverse().ToArray();
+                var length = BitConverter.ToUInt16(buffer);
+
+                buffer = new byte[length];
+                var read = await from.ReceiveAsync(buffer, cancellationToken);
+                if (read != buffer.Length) {
+                    _sawmill.Warning("failed to parse byond topic due to insufficient data read");
+                    return null;
+                }
+
+                return Encoding.ASCII.GetString(buffer[5..^1]);
+            }
+
+            if (_worldTopicSocket is null)
+                throw new InvalidOperationException("Attempted to start the World Topic Listener without a valid socket bind address.");
+
+            while (!cancellationToken.IsCancellationRequested) {
+                try {
+                    var pending = await _worldTopicSocket.AcceptAsync(cancellationToken);
+                    var topic = await ParseByondTopic(pending);
```

Comment:
TGS won't send simultaneous topics but it's still probably a case you want to handle. Especially since a sleep() in world/Topic should block the entire response. Everything after the Accept should be done asynchronously without being immediately awaited so further accepts can be processed simultaneously.